src/fast_graph_builder.rs

```diff
@@ -169,19 +169,14 @@ impl FastGraphBuilder {
             }
             self.fast_graph.first_edge_ids_bwd[rank + 1] = self.fast_graph.get_num_in_edges();
 
-            self.fast_graph.ranks[rank] = node;
+            self.fast_graph.ranks[node] = rank;
             node_contractor::contract_node(&mut preparation_graph, &mut dijkstra, node);
             //            println!("contracted node {} / {}, num edges fwd: {}, num edges bwd: {}", rank+1, self.num_nodes, self.fast_graph.get_num_out_edges(), self.fast_graph.get_num_in_edges());
         }
         self.finish_contraction();
     }
 
     fn finish_contraction(&mut self) {
-        let ranks_copy = self.fast_graph.ranks.clone();
-        for i in 0..ranks_copy.len() {
```

Comment:
Nice we do not have to do this. This is unrelated to stall on demand though, right? Can you open a separate PR for this to master? Otherwise I will just commit this straight to master.

Reply:
I commited it to master here: https://github.com/easbar/fast_paths/commit/192ae1997f9857791826ac5ed16892b2f692920c